File: Java/OOP/Inheritance/Hierarchical inheritance  (example 1).java
class Institute {
    String institute_name = "Institute of Engineering"; // just a random name
}

class Teacher extends Institute {
    String teacher_name = "James Gosling";

    void info() {
        System.out.println(teacher_name + " works as a teacher at " + institute_name);
    }
}

class Student extends Institute {
    String student_name = "Abdullah Al Naiem";

    void info() {
        System.out.println(student_name + " studying at " + institute_name);
    }
}

class hierarchicalInhertance {
    public static void main(String[] args) {
        Teacher teacher = new Teacher();
        teacher.info();
        Student student = new Student();
        student.info();
    }
}

File: Python/OOP/Inheritance/Hierarchical inheritance (example 1).py
class Institute:
    _institute_name = "Institute of Engineering"  # just a random name


class Teacher(Institute):
    teacher_name = "Guido van Rossum"

    def info(self):
        print(self.teacher_name + " works as a teacher at " + self._institute_name)


class Student(Institute):
    student_name = "Abdullah Al Naiem"

    def info(self):
        print(self.student_name + " studying at " + self._institute_name)


class hierarchicalInhertance:
    teacher = Teacher()
    teacher.info()
    student = Student()
    student.info()